lnwire/update_add_htlc.go

```diff
@@ -152,16 +185,51 @@ func (c *UpdateAddHTLC) Encode(w *bytes.Buffer, pver uint32) error {
 		return err
 	}
 
-	// Only include blinding point in extra data if present.
+	// Construct a slice of all the records that we should include in the
+	// extra data field.
 	var records []tlv.RecordProducer
 
+	// Only include blinding point in extra data if present.
 	c.BlindingPoint.WhenSome(func(b tlv.RecordT[BlindingPointTlvType,
 		*btcec.PublicKey]) {
 
 		records = append(records, &b)
 	})
 
-	err := EncodeMessageExtraData(&c.ExtraData, records...)
+	// Extend the 'records' slice with TLV records from the custom records
+	// field.
+	//
+	// As a sanity check, we will first ensure that the existing records
+	// TLV types are less than the minimum custom records TLV type.
+	for _, recordProducer := range records {
+		record := recordProducer.Record()
+		recordTlvType := uint64(record.Type())
+
+		if recordTlvType >= MinCustomRecordsTlvType {
+			return fmt.Errorf("records contain TLV type that is "+
+				"unexpectedly greater than or equal to the "+
+				"minimum TLV type of custom records: %d",
+				recordTlvType)
+		}
+	}
+
+	// Validate custom records field. This will ensure that all custom
+	// records are within the correct TLV type range.
+	err := c.CustomRecords.Validate()
+	if err != nil {
+		return err
+	}
+
+	// Convert the custom records map to TLV records and append them to the
+	// exiting records slice.
+	recordsA := tlv.MapToRecords(c.CustomRecords)
+	for _, record := range recordsA {
+		r := recordProducer{record}
+		records = append(records, &r)
+	}
+
+	// Encode the records into the extra data field.
+	err = EncodeMessageExtraData(&c.ExtraData, records...)
```

Comment:
Let's add some unit tests for this.

Reply:
I've added `UpdateAddHTLC` message encoding/decoding unit test. Let me know if that's not what you had in mind. Or if you think it makes sense to add any extra test cases.